core/services/ocr2/plugins/ocr2keeper/evm21/encoder.go

```diff
@@ -230,9 +159,20 @@ func (enc EVMAutomationEncoder21) KeysFromReport(b []byte) ([]ocr2keepers.Upkeep
 	return keys, nil
 }
 
-type wrappedTrigger struct {
-	BlockNumber uint32   `abi:"blockNumber"`
-	BlockHash   [32]byte `abi:"blockHash"`
+// validateReport checks that the report is valid, currently checking that all
+// lists are the same length.
+// TODO: add more validations? e.g. parse validate triggers
+func (enc EVMAutomationEncoder21) validateReport(report automation_utils_2_1.KeeperRegistryBase21Report) error {
```

Comment:
curious where this is used? I think it would be safe to assume all signed reports will be valid

Reply:
OK keeping the lists length checking to avoid nil reference error